# exlude posix time_exploded and time_now from mac platform and add task_scheduler/environment_config

--- a/tools/gn/bootstrap/bootstrap.py.orig	2017-08-22 12:29:55.000000000 +0300
+++ b/tools/gn/bootstrap/bootstrap.py	2017-08-22 12:30:07.000000000 +0300
@@ -570,6 +570,12 @@
       'base/vlog.cc',
   ])

+  if is_posix and not is_mac:
+    static_libraries['base']['sources'].extend([
+        'base/time/time_exploded_posix.cc',
+        'base/time/time_now_posix.cc',
+    ])
+
   if is_posix:
     static_libraries['base']['sources'].extend([
         'base/base_paths_posix.cc',
@@ -597,13 +603,12 @@
         'base/synchronization/waitable_event_posix.cc',
         'base/sys_info_posix.cc',
         'base/task_scheduler/task_tracker_posix.cc',
+        'base/task_scheduler/environment_config.cc',
         'base/threading/platform_thread_internal_posix.cc',
         'base/threading/platform_thread_posix.cc',
         'base/threading/thread_local_storage_posix.cc',
         'base/threading/worker_pool_posix.cc',
         'base/time/time_conversion_posix.cc',
-        'base/time/time_exploded_posix.cc',
-        'base/time/time_now_posix.cc',
         'base/trace_event/heap_profiler_allocation_register_posix.cc',
     ])
     static_libraries['libevent'] = {
